### ISSUE TITLE
Exception when trying to check the status of a starting server in a while loop

### ISSUE THREAD

**Issue by Reviville** [Dinnerbone/mcstatus#53]
Hey ho,

as the title already mentions, im trying to check the status of a starting minecraft server using a while-loop with `server.status()`

My log then shows the following exception:

`AttributeError: 'TCPSocketConnection' object has no attribute 'socket'
Exception ignored in: <bound method TCPSocketConnection.__del__ of <mcstatus.protocol.connection.TCPSocketConnection object at 0x7fb4005e1e10>>
Traceback (most recent call last):
  File "/path/to/project/.venv/lib/python3.6/site-packages/mcstatus/protocol/connection.py", line 153, in __del__
    self.socket.close()`

If the server is up, it works. So it works if you request the server a single time, just not when trying to request it several times in a loop.
Any suggestions?

**Comment by MarkL4YG** [comment]
Seems like the problem is that there's no socket to close when the connection failed or has been closed already...  
For now try initializing a new Server each time you want to check the status and see if that works as wanted.

**Comment by MarkL4YG** [comment]
@Reviville Does inactivity mean you resolved your issue or you gave up on it?

**Comment by Reviville** [comment]
@MarkL4YG i gave up on it as this error still occurs.

**Comment by MarkL4YG** [comment]
You may try going into [mcstatus/protocol/connection.py l182](https://github.com/Dinnerbone/mcstatus/blob/master/mcstatus/protocol/connection.py#L182) and replace
```python
    def __del__(self):
        self.socket.close()
```
with
```python
    def __del__(self):
        try:
            self.socket.close()
        except:
            pass
```
and test if the application still halts.

This by no means prevents the error, it should just pass the exception allowing the application to continue running ___as if___ nothing happened.

**Comment by Reviville** [comment]
Just did.
Seems to "work" for now.
Thank you.

**Comment by MarkL4YG** [comment]
@Reviville FYI The exception indicates that something in the connection failed prior to setting the socket causing a common NullPointerException when the connection is being destroyed for garbage collection.
Using such a fix should therefore not be any problem since a nonexistent socket doesn't require to be closed obviously.

**Comment by kevinkjt2000** [comment]
Fixed in #56
